addons/event/static/src/client_action/event_barcode.js

```diff
@@ -0,0 +1,111 @@
+import { Component, onWillStart } from "@odoo/owl";
+import { registry } from "@web/core/registry";
+import { useBus, useService } from "@web/core/utils/hooks";
+import { EventRegistrationSummaryDialog } from "./event_registration_summary_dialog";
+
+export class EventScanView extends Component {
+    setup() {
+        this.actionService = useService("action");
+        this.dialog = useService("dialog");
+        this.notification = useService("notification");
+        this.orm = useService("orm");
+        this.rpc = useService("rpc");
+
+        const { default_event_id, active_model, active_id } = this.props.action.context;
+        this.eventId = default_event_id || (active_model === "event.event" && active_id);
+        this.isMultiEvent = !this.eventId;
+
+        const barcode = useService("barcode");
+        useBus(barcode.bus, "barcode_scanned", (ev) => this.onBarcodeScanned(ev.detail.barcode));
+
+        onWillStart(this.onWillStart);
+    }
+
+    /**
+     * @override
+     * Fetch barcode init information. Notably eventId triggers mono- or multi-
+     * event mode (Registration Desk in multi event allow to manage attendees
+     * from several events and tickets without reloading / changing event in UX.
+     */
+    async onWillStart() {
+        this.data = await this.rpc("/event/init_barcode_interface", {
+            event_id: this.eventId,
+        });
+    }
+
+    /**
+     * When scanning a barcode, call Registration.register_attendee() to get
+     * formatted registration information, notably its status or event-related
+     * information. Open a confirmation / choice Dialog to confirm attendee.
+     */
+    async onBarcodeScanned(barcode) {
+        const result = await this.orm.call("event.registration", "register_attendee", [], {
+            barcode: barcode,
+            event_id: this.eventId,
+        });
+
+        if (result.error && result.error === "invalid_ticket") {
+            this.notification.add(_t("Invalid ticket"), {
+                title: _t("Warning"),
+                type: "danger",
+            });
+        } else {
+            this.registrationId = result.id;
+            this.dialog.add(EventRegistrationSummaryDialog, { registration: result });
+        }
+    }
+
+    onClickSelectAttendee() {
+        if (this.isMultiEvent) {
+            this.actionService.doAction("event.event_registration_action");
+        } else {
+            this.actionService.doAction("event.event_registration_action_kanban", {
+                additionalContext: {
+                    active_id: this.eventId,
+                    search_default_unconfirmed: true,
+                    search_default_confirmed: true,
+                },
+            });
+        }
+    }
+
+    onClickBackToEvents() {
+        if (this.isMultiEvent) {
+            // define action from scratch instead of using existing 'action_event_view' to avoid
+            // messing with menu bar
+            this.actionService.doAction({
```

Comment:
You can probably use the 'action_event_view' action that exists, unless you would have reason not to do so ? that way we have less custom action and we rely on existing xml records ...

Reply:
@tde-banana-odoo i tried it but then it puts me into this wierd view
`this.actionService.doAction("event.action_event_view");`

![image](https://github.com/odoo/odoo/assets/52849166/6928bd0b-90ae-413e-84c1-cc52e4da58a2)

Reply:
i mean there are no menus available at the top

Reply:
refreshing helps but i don't think that's a correct flow
![image](https://github.com/odoo/odoo/assets/52849166/4fc9d13c-35e3-4993-9ec3-3ad84148024d)

Reply:
i guess that's what the comment was saying, so i will leave this one